pymongo/command_cursor.py

```diff
@@ -149,13 +149,14 @@ def duration(): return datetime.datetime.now() - start
         reply = response.data
 
         try:
-            docs = self._unpack_response(reply,
-                                         self.__id,
-                                         self.__collection.codec_options)
-            if from_command:
-                first = docs[0]
-                client._receive_cluster_time(first, self.__session)
-                helpers._check_command_response(first)
+            with client._reset_on_error(self.__address, self.__session):
```

Comment:
_reset_on_error catches and resets on OperationFailure, but this previously avoided resetting on OperationFailure. Is this change correct?

Reply:
Yes, we are supposed to reset on certain OperationFailures (like ShutdownInProgress). I think it was an bug that we did not. See https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#not-master-and-node-is-recovering

I should probably make this change in a new ticket so we can include it in 3.8 as well.

Reply:
SGTM. No one has complained about this either so I don't think we will need to do backport. 🤞

Reply:
See https://jira.mongodb.org/browse/PYTHON-1759.